lib/PuppeteerSharp/LifecycleWatcher.cs

```diff
@@ -70,8 +72,15 @@ internal class LifecycleWatcher : IDisposable
         public Task<bool> SameDocumentNavigationTask => _sameDocumentNavigationTaskWrapper.Task;
         public Task<bool> NewDocumentNavigationTask => _newDocumentNavigationTaskWrapper.Task;
         public Response NavigationResponse => _navigationRequest?.Response;
-        public Task TimeoutOrTerminationTask
-            => _terminationTaskWrapper.Task.WithTimeout(_timeout);
+        public Task TimeoutOrTerminationTask
+        {
+            get
+            {
+                var task = _terminationTaskWrapper.Task.WithTimeout(_timeout);

```

Comment:
Can we try to find a solution so we can solve this inside the `WithTimeout` method?
This is not the only place where we use `WithTimeout`. We are using it in the `Page` class, in `Browser` and in `ChromumProcess` among others. 
I can buy if we need to create a kind of TimeoutTask class or something like this.

Reply:
Sure, that would be a cleaner solution. I don't think we need a new class, just a way to ensure we don't create new tasks on every property get.

Reply:
BTW, Great Job!